src/pycram/designators/action_designator.py

```diff
@@ -783,6 +783,134 @@ def description(cls, object_designator: Union[Iterable[Object], Object],
         return PartialDesignator(TransportAction, object_designator=object_designator,
                                  target_location=target_location,
                                  arm=arm)
+    
+
+from dataclasses import dataclass
+from pycram.designators.action_designator import ActionDescription, with_plan
+from pycram.designators.object_designator import ObjectDesignatorDescription
+from pycram.designators.location_designator import CostmapLocation
+from pycram.designators.action_designator import (
+    PickUpActionDescription,
+    PlaceActionDescription,
+    ParkArmsActionDescription
+)
+from pycram.datastructures.enums import Arms
+from pycram.world_concepts.belief_world import BelieveObject
+from pycram.robot_descriptions import RobotDescription
+from pycram.errors import ConfigurationNotReached, ReachabilityFailure
+from pycram.datastructures.pose import PoseStamped
+from pycram.designators import PartialDesignator
+
+
+@with_plan
+@dataclass
+class EfficientTransportAction(ActionDescription):
+    """
+    This action transports an object to a target location by choosing the closest
+    available arm using simple Euclidean distance.
+    """
+    object_designator: ObjectDesignatorDescription
+    target_location: PoseStamped
+
+    def plan(self) -> None:
+        # Get the robot and object from the belief world
+        robot = BelieveObject(names=[RobotDescription.current_robot_description.name]).resolve()
+        obj = self.object_designator.resolve()
+
+        if not obj or not obj.pose:
+            raise ConfigurationNotReached(f"Couldn't resolve the pose for the object: {self.object_designator}")
+
+        object_position = obj.pose.position  # (x, y, z)
+
+        # The gripper positions for arms
+        left_tip = robot.get_link_position("left_gripper")
```

Comment:
This only works for tracy since the names are different for other robots. Please use the RobotDescription to get the tool_frames